bug: Vue - Tabs - 5.9.2 - Double click on TabButton breaks tabs-routing

### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

When double clicking a Tab, routing to other Tabs when pressing their TabButton is not working anymore. See:

https://user-images.githubusercontent.com/24892172/145353512-2df35d16-51cb-4b86-9ce2-b14a0661821e.mov




### Expected Behavior

Expected to be routed to the Tab when pressing on the Tab2-TabButton (Step 4).

### Steps to Reproduce

1. Click on Tab2 --> Routes to Tab2
2. Click on Tab1 --> Routes to Tab1
3. Click Tab1 again --> nothing happens (we already are on the correct tab)
4. Click Tab2 --> route changes but we are not routed to Tab2

Reprocution code is simply a Tabs starter project generated with:
```
ionic start ionic-vue-tabs-double-tab-bug tabs --type vue
```

and changed these two lines in the package.json to:
```
    "@ionic/vue": "5.9.2",
    "@ionic/vue-router": "5.9.2",
```

### Code Reproduction URL

https://github.com/Predarion/ionic-vue-tabs-double-tab-bug

### Ionic Info

[WARN] Error loading @capacitor/ios package.json: Error: Cannot find module '@capacitor/ios/package'

       Require stack:
       - /usr/local/lib/node_modules/@ionic/cli/lib/project/index.js
       - /usr/local/lib/node_modules/@ionic/cli/lib/index.js
       - /usr/local/lib/node_modules/@ionic/cli/index.js
       - /usr/local/lib/node_modules/@ionic/cli/bin/ionic
[WARN] Error loading @capacitor/android package.json: Error: Cannot find module '@capacitor/android/package'

       Require stack:
       - /usr/local/lib/node_modules/@ionic/cli/lib/project/index.js
       - /usr/local/lib/node_modules/@ionic/cli/lib/index.js
       - /usr/local/lib/node_modules/@ionic/cli/index.js
       - /usr/local/lib/node_modules/@ionic/cli/bin/ionic

Ionic:

   Ionic CLI       : 6.18.1 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 5.9.2

Capacitor:

   Capacitor CLI      : 3.3.3
   @capacitor/android : not installed
   @capacitor/core    : 3.3.3
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.5.0

System:

   NodeJS : v16.13.0 (/usr/local/bin/node)
   npm    : 8.1.0
   OS     : macOS Big Sur


### Additional Information

This bug might be introduced with the changes from

* **vue:** switching between tabs preserves query string ([#24297](https://github.com/ionic-team/ionic/issues/24297)) ([047d3c7](https://github.com/ionic-team/ionic/commit/047d3c77729db08e4fd84f426f6c5c2af0eacc52)), closes [#23699](https://github.com/ionic-team/ionic/issues/23699)

as this behavior does not occur with version 5.9.1, but does occur with version 5.9.2 and the dev build of aboves issue: 

```
npm install @ionic/vue@5.10.0-dev.202112011701.ac1402b @ionic/vue-router@5.10.0-dev.202112011701.ac1402b
```

--- a/packages/vue/src/components/IonTabBar.ts
+++ b/packages/vue/src/components/IonTabBar.ts
@@ -113,7 +113,14 @@ export const IonTabBar = defineComponent({
          * land on /tabs/tab1/child instead of /tabs/tab1.
          */
         if (activeTab !== prevActiveTab || (prevHref !== currentRoute.pathname)) {
-          const search = (currentRoute.search !== undefined) ? `?${currentRoute.search}` : '';
+
+          /**
+           * By default the search is `undefined` in Ionic Vue,
+           * but Vue Router can set the search to the empty string.
+           * We check for truthy here because empty string is falsy
+           * and currentRoute.search cannot ever be a boolean.
+           */
+          const search = (currentRoute.search) ? `?${currentRoute.search}` : '';
           tabs[activeTab] = {
             ...tabs[activeTab],
             currentHref: currentRoute.pathname + search

--- a/packages/vue/test-app/tests/e2e/specs/tabs.js
+++ b/packages/vue/test-app/tests/e2e/specs/tabs.js
@@ -305,7 +305,30 @@ describe('Tabs', () => {
     cy.ionPageHidden('tab2');
 
     cy.url().should('include', '/tabs/tab1/child-one?key=value');
-  })
+  });
+
+  // Verifies fix for https://github.com/ionic-team/ionic-framework/issues/23699
+  it('should handle clicking tab multiple times without query string', () => {
+    cy.visit('http://localhost:8080/tabs/tab1');
+
+    cy.ionPageVisible('tab1');
+
+    cy.get('ion-tab-button#tab-button-tab2').click();
+    cy.ionPageVisible('tab2');
+    cy.ionPageHidden('tab1');
+
+    cy.get('ion-tab-button#tab-button-tab1').click();
+    cy.ionPageVisible('tab1');
+    cy.ionPageHidden('tab2');
+
+    cy.get('ion-tab-button#tab-button-tab1').click();
+    cy.ionPageVisible('tab1');
+    cy.ionPageHidden('tab2');
+
+    cy.get('ion-tab-button#tab-button-tab2').click();
+    cy.ionPageVisible('tab2');
+    cy.ionPageHidden('tab1');
+  });
 })
 
 describe('Tabs - Swipe to Go Back', () => {

